paymentsheet/src/main/java/com/stripe/android/paymentsheet/AddPaymentMethodsAdapter.kt

```diff
@@ -74,42 +75,122 @@ internal class AddPaymentMethodsAdapter(
-    ) : RecyclerView.ViewHolder(binding.root) {
-        constructor(parent: ViewGroup) : this(
-            LayoutPaymentsheetAddPaymentMethodCardViewBinding.inflate(
-                LayoutInflater.from(parent.context), parent, false
-            )
+        private val composeView: ComposeView,
+        private val viewWidth: Dp
+    ) : RecyclerView.ViewHolder(composeView) {
+
+        constructor(parent: ViewGroup, numberOfPaymentMethods: Int) : this(
+            composeView = ComposeView(parent.context).apply {
+                ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed
+            },
+            viewWidth = calculateViewWidth(parent, numberOfPaymentMethods)
         )
 
-        fun setSelected(selected: Boolean) {
-            binding.root.isSelected = selected
-            binding.root.strokeWidth = if (selected) {
-                itemView.resources
-                    .getDimension(R.dimen.stripe_paymentsheet_add_pm_card_stroke_width_selected)
-                    .roundToInt()
-            } else {
-                itemView.resources
-                    .getDimension(R.dimen.stripe_paymentsheet_add_pm_card_stroke_width)
-                    .roundToInt()
-            }
-            binding.root.elevation = if (selected) {
-                itemView.resources
-                    .getDimension(R.dimen.stripe_paymentsheet_add_pm_card_elevation_selected)
-            } else {
-                itemView.resources
-                    .getDimension(R.dimen.stripe_paymentsheet_add_pm_card_elevation)
+        fun bind(
+            paymentMethod: SupportedPaymentMethod,
+            isSelected: Boolean,
+            isEnabled: Boolean,
+            onItemSelectedListener: () -> Unit
+        ) {
+            composeView.setContent {
+                PaymentMethodUI(
+                    viewWidth = viewWidth,
+                    iconRes = paymentMethod.iconResource,
+                    title = itemView.resources.getString(paymentMethod.displayNameResource),
+                    isSelected = isSelected,
+                    isEnabled = isEnabled,
+                    onItemSelectedListener = onItemSelectedListener
+                )
             }
         }
 
-        fun setEnabled(enabled: Boolean) {
-            binding.root.isEnabled = enabled
-            binding.title.isEnabled = enabled
+        fun onViewRecycled() {
+            // Dispose the underlying Composition of the ComposeView
+            // when RecyclerView has recycled this ViewHolder
+            composeView.disposeComposition()
         }
+    }
+
+    internal companion object {
+        private fun calculateViewWidth(parent: ViewGroup, numberOfPaymentMethods: Int): Dp {
+            val targetWidth = parent.measuredWidth - parent.paddingStart - parent.paddingEnd
+            val screenDensity = parent.context.resources.displayMetrics.density
+            val minItemWidth = 100 * screenDensity + 6.0f + 6.0f
```

Comment:
Where did we get these numbers from? Can we describe these numbers so it is more readable?

Reply:
fixed this with some constants.